find_leaves_of_binary_tree.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+
+
+import sys
+sys.path.insert(0, '../data_structure')
+from data_structure.TreeNode import TreeNode
+import collections
+
+class Myclass:
+	"""
+	Given a binary tree, find all leaves and then remove those leaves. Then repeat the previous steps until the tree is empty.
+
+Example:
+Given binary tree 
+          1
+         / \
+        2   3
+       / \     
+      4   5    
+Returns [4, 5, 3], [2], [1].
+
+Explanation:
+1. Remove the leaves [4, 5, 3] from the tree
+
+          1
+         / 
+        2          
+2. Remove the leaf [2] from the tree
+
+          1          
+3. Remove the leaf [1] from the tree
+
+          []         
+Returns [4, 5, 3], [2], [1].
+
+	"""
+	def find_all_leaves(self, root):
+		"""
+		:type root: TreeNode
+		:rtype [[TreeNode]]
+		"""
+		ret = collections.defaultdict(list)
+		if not root:
+			return ret
+		self.find_all_leaves_helper(root, ret)
+		return ret.values()
+
+	def find_all_leaves_helper(self, root, ret):
+		"""
+		:type root: TreeNode
+		:type ret: {}
+		:rtype: int
+		"""
+		if not root:
+			return -1
+		level = 1 + max(self.find_all_leaves_helper(root.left, ret), self.find_all_leaves_helper(root.right, ret))
```

Comment:
I am not sure this line is longer than 80 characters or not, but seems like too long for me.

       level = 1 + max(
           self.find_all_leaves_helper(root.left, ret),  #<--four spaces indent 
           self.find_all_leaves_helper(root.right, ret))